airsonic-main/src/main/java/org/airsonic/player/service/search/IndexType.java

```diff
@@ -0,0 +1,209 @@
+/*
+ This file is part of Airsonic.
+
+ Airsonic is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Airsonic is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Airsonic.  If not, see <http://www.gnu.org/licenses/>.
+
+ Copyright 2016 (C) Airsonic Authors
+ Based upon Subsonic, Copyright 2009 (C) Sindre Mehus
+ */
+
+package org.airsonic.player.service.search;
+
+import java.util.AbstractMap;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Enum that symbolizes the each lucene index entity.
+ * This class is a division of what was once part of SearchService and added functionality.
+ * @since legacy
+ */
+public enum IndexType {
+
+    /*
+     * Boosts is a factor for search scores, which is 1 by default.
+     */
+    SONG(
+        fieldNames(
+            FieldNames.TITLE,
+            FieldNames.ARTIST),
+        boosts(
+            entry(FieldNames.TITLE, 2F))),
+
+    ALBUM(
+        fieldNames(
+            FieldNames.ALBUM,
+            FieldNames.ARTIST,
+            FieldNames.FOLDER),
+        boosts(
+            entry(FieldNames.ALBUM, 2F))),
+
+    ALBUM_ID3(
+        fieldNames(
+            FieldNames.ALBUM,
+            FieldNames.ARTIST,
+            FieldNames.FOLDER_ID),
+        boosts(
+            entry(FieldNames.ALBUM, 2F))),
+
+    ARTIST(
+        fieldNames(
+            FieldNames.ARTIST,
+            FieldNames.FOLDER),
+        boosts(
+            entry(FieldNames.ARTIST, 1F))),
+
+    ARTIST_ID3(
+        fieldNames(
+            FieldNames.ARTIST),
+        boosts(
+            entry(FieldNames.ARTIST, 2F))),
+
+    ;
+
+  public static final class FieldNames {
+
+    private FieldNames() {

```

Comment:
Is this necessary?

Reply:
It is necessary, but it is difficult to answer immediately if it is said about the description place and scope.
"Don't put it in a public enum if you can"
Is this all right?

Reply:
It used to exist as a private FIELD_TITLE in SearchService.
As a result of the division, it was referred to from here and there.

I'll check if package access is possible.
If possible, exclude from enum.